File: test/se/crisp/signup4/acceptance/pages/SignUpPage.java
package se.crisp.signup4.acceptance.pages;

import se.crisp.signup4.acceptance.PlayContainer;
import se.crisp.signup4.acceptance.SharedDriver;
import org.junit.Assert;
import org.openqa.selenium.By;

public class SignUpPage {

  private final SharedDriver driver;

  public SignUpPage(SharedDriver driver) {
    this.driver = driver;
  }

  public SignUpPage navigateTo(long userId, long eventId) {
    driver.navigate().to(PlayContainer.getBaseUrl() + "/participations/edit?eventId=" + eventId + "&userId=" + userId);
    Assert.assertEquals("Anmälan", driver.findElement(By.id("page_name")).getText());
    return this;
  }

  public SignUpPage setStatus(String status) {
    driver.findElement(By.id(status.toLowerCase())).click();
    return this;
  }

  public SignUpPage save() {
    driver.findElement(By.id("action")).submit();
    return this;
  }

  public SignUpPage addComment(String comment) {
    driver.findElement(By.id("comment")).sendKeys(comment);
    return this;
  }
}

File: test/se/crisp/signup4/acceptance/pages/SignUpPage.scala
package se.crisp.signup4.acceptance.pages

import org.openqa.selenium.{By, WebDriver}

class SignUpPage(baseUrl: String, driver: WebDriver) extends Page(baseUrl, driver) {

  def navigateTo(userId: Long, eventId: Long) {
    driver.navigate.to(baseUrl + "/participations/edit?eventId=" + eventId + "&userId=" + userId)
    waitForCompletePageLoad()
  }

  def setStatus(status: String) {
    driver.findElement(By.id(status.toLowerCase)).click()
  }

  def save() {
    driver.findElement(By.id("action")).submit()
    waitForCompletePageLoad()
  }

  def addComment(comment: String) {
    driver.findElement(By.id("comment")).sendKeys(comment)
  }
}
